src/ralph/models/converter.py

```diff
@@ -0,0 +1,82 @@
+"""Base converter definition"""
+
+from abc import ABC, abstractmethod
+from uuid import UUID, uuid5
+
+from pydantic import BaseModel
+
+from ralph.defaults import MODEL_PATH_SEPARATOR
+from ralph.exceptions import ConversionException
+from ralph.utils import get_dict_value_from_path, set_dict_value_from_path
+
+
+class BaseConverter(ABC):
+    """Converter Base Class.
+
+    Converters define a conversion set to convert from one event type to another.
+    """
+
+    __model__: BaseModel
+
+    def __init__(self, uuid_namespace: UUID):
+        """Initializes BaseConverter."""
+
+        self.uuid_namespace = uuid_namespace
+        self.field_array = []
+        self.prepare_converter(self.field_array, self.get_conversion_set())
+
+    @abstractmethod
+    def get_conversion_set(self):
+        """Returns a conversion set used for conversion.
+
+        Each element in the conversion set should be a tuple of length 2 or 3.
+        - The first element of each tuple is a destination path string.
+        - The second element can be a source path string or a function with no arguments.
+        - The optional third element defines a transformer function with one argument.
+        """
+
+    def convert(self, event: dict, event_str: str):
```

Comment:
Maybe we have a design issue here: providing both the model instance and its source string is not that elegant. It's only used to calculate a UUID given a string, so maybe we should delegate this elsewhere (not in the converter), and provide a UUID to the `convert` method instead? Or should we embed source string to our event models?

Reply:
You are right) Thanks) Removed the `uuid_namespace` object field and replaced the convert method argument `event_str` with `uuid`)